presto-spi/src/test/java/com/facebook/presto/spi/block/BenchmarkFloorMod.java

```diff
@@ -0,0 +1,185 @@
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jmh.runner.options.VerboseMode;
+
+import java.util.Random;
+import java.util.concurrent.ThreadLocalRandom;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+@SuppressWarnings("MethodMayBeStatic")
+@State(Scope.Thread)
+@OutputTimeUnit(MILLISECONDS)
+@Fork(3)
+@Warmup(iterations = 10, time = 500, timeUnit = MILLISECONDS)
+@Measurement(iterations = 10, time = 500, timeUnit = MILLISECONDS)
+@BenchmarkMode(Mode.AverageTime)
+public class BenchmarkFloorMod
+{
+    static final int numHashCodes = 1000000;
+    int hashTableSize = ThreadLocalRandom.current().nextInt() & 0x7fff_ffff;
+
+    @Benchmark
+    public void ReduceHashPosition1(RandBenchmarkData data, Blackhole blackhole)
+            throws Throwable
+    {
+        long[] hashcodes = data.getHashcodes();
+
+        for (long i : hashcodes) {
+            blackhole.consume(getReducedHashPosition1(i, hashTableSize));
+        }
+    }
+
+    @Benchmark
+    public void ReduceHashPosition2(RandBenchmarkData data, Blackhole blackhole)
+            throws Throwable
+    {
+        long[] hashcodes = data.getHashcodes();
+
+        for (long i : hashcodes) {
+            blackhole.consume(getReducedHashPosition2(i, hashTableSize));
+        }
+    }
+
+    @Benchmark
+    public void ReduceHashPosition3(RandBenchmarkData data, Blackhole blackhole)
+            throws Throwable
+    {
+        long[] hashcodes = data.getHashcodes();
+
+        for (long i : hashcodes) {
+            blackhole.consume(getReducedHashPosition3(i, hashTableSize));
+        }
+    }
+
+    @Benchmark
+    public void ReduceHashPosition4(RandBenchmarkData data, Blackhole blackhole)
+            throws Throwable
+    {
+        long[] hashcodes = data.getHashcodes();
+
+        for (long i : hashcodes) {
+            blackhole.consume(getReducedHashPosition4(i, hashTableSize));
+        }
+    }
+
+    @Benchmark
+    public void ReduceHashPosition5(RandBenchmarkData data, Blackhole blackhole)
+            throws Throwable
+    {
+        long[] hashcodes = data.getHashcodes();
+
+        for (long i : hashcodes) {
+            blackhole.consume(getReducedHashPosition5(i, hashTableSize));
+        }
+    }
+
+    @Benchmark
+    public void floorMod(RandBenchmarkData data, Blackhole blackhole)
+            throws Throwable
+    {
+        long[] hashcodes = data.getHashcodes();
+        for (long i : hashcodes) {
+            blackhole.consume(Math.floorMod(i, hashTableSize));
+        }
+    }
+
+    @SuppressWarnings("FieldMayBeFinal")
+    @State(Scope.Thread)
+    public static class RandBenchmarkData
+    {
+        private Random random;
+        long[] hashcodes = new long[numHashCodes];
+
+        @Setup
+        public void setup()
+                throws Exception
+        {
+            random = new Random(0);
+            for (int i = 0; i < numHashCodes; i++) {
+                hashcodes[i] = random.nextLong();
```

Comment:
Any reason why not using `ThreadLocalRandom.current().nextLong()` ?

Reply:
changed